examples/dtls-sock/main.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       DTLS sock example
+ *
+ * @author      Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ */
+
+#include <stdio.h>
+
+#include "shell.h"
+#include "msg.h"
+#include "dtls.h"
+#include "net/gnrc.h"
+#include "net/gnrc/netif.h"
+
+#define MAIN_QUEUE_SIZE         (8)
+static msg_t _main_msg_queue[MAIN_QUEUE_SIZE];
+
+extern int dtls_client_cmd(int argc, char **argv);
+extern int dtls_server_cmd(int argc, char **argv);
+
+static const shell_command_t shell_commands[] = {
+    { "dtlsc", "Start a DTLS client", dtls_client_cmd },
+    { "dtlss", "Start and stop a DTLS server", dtls_server_cmd },
+    { NULL, NULL, NULL }
+};
+
+int main(void)
+{
+    /* we need a message queue for the thread running the shell in order to
+     * receive potentially fast incoming networking packets */
```

Comment:
Where is this comment coming from? The message queue is only needed for commands like `ping6` that rely on `gnrc_netapi`. `sock`s as you know rely on `mbox` so don't require the queue.

Reply:
I copied this from the dtls-echo example but other examples such as gnrc_networking, cord_ep, gcoap (cord_ep and gcoap also uses `sock`) also have a similar comment. So maybe it is all copied and pasted from somewhere. Documentation about when the queue is needed somewhere would be nice.

Tested without the queue and it still works. I'll remove the queue.

Reply:
You are right... And technically the comment is also right, just misleading.

The percondition for using `gnrc_netapi` with a thread is documented [here](https://doc.riot-os.org/group__net__gnrc__netreg.html#gafb6a9d0b840aae07c3c1617932835ef1).